src/irr/asset/CImageLoaderPNG.cpp

```diff
@@ -252,36 +243,21 @@ asset::SAssetBundle CImageLoaderPng::loadAsset(io::IReadFile* _file, const asset
         return {};
 	}
 
-	// OpenGL cannot transfer rows with arbitrary padding
-    static const uint32_t MAX_PITCH_ALIGNMENT = 8u;
-    // try with largest alignment first
-    auto calcPitchInBlocks = [](uint32_t width, uint32_t blockByteSize) -> uint32_t
-    {
-		auto rowByteSize = width*blockByteSize;
-		for (uint32_t _alignment=MAX_PITCH_ALIGNMENT; _alignment>1u; _alignment>>=1u)
-		{
-			auto paddedSize = core::alignUp(rowByteSize, _alignment);
-			if (paddedSize % blockByteSize)
-				continue;
-			return paddedSize/blockByteSize;
-		}
-        return width;
-    };
-
-    const uint32_t texelFormatBytesize = getTexelOrBlockBytesize(image->getCreationParameters().format);
+    const uint32_t texelFormatBytesize = getTexelOrBlockBytesize(imgInfo.format);
 
-    auto texelBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(image->getImageDataSizeInBytes());
     auto regions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>>(1u);
     ICPUImage::SBufferCopy& region = regions->front();
     //region.imageSubresource.aspectMask = ...; //waits for Vulkan
     region.imageSubresource.mipLevel = 0u;
     region.imageSubresource.baseArrayLayer = 0u;
     region.imageSubresource.layerCount = 1u;
     region.bufferOffset = 0u;
-    region.bufferRowLength = calcPitchInBlocks(Width, texelFormatBytesize);
+    region.bufferRowLength = asset::IImageAssetHandlerBase::calcPitchInBlocks(Width, texelFormatBytesize);
```

Comment:
you should assert that block size == 1x1x1

Reply:
why? it's png loader, the dim won't be different from 1x1x1

Reply:
ASSERT

Reply:
done